### PR TITLE
Feature :: influxDB integration

### LINKED ISSUE #2564: Feature :: Tooljet integration for InfluxDB 🚀
### Is your proposal related to a problem?
Currently ToolJet doesn't have InfluxDB integration.

### Describe the solution you'd like
Integration for InfluxDB.

**Supported Operations**

- [x] List buckets

- [x] Create a bucket

- [x] Delete a bucket

- [x] Retrieve a bucket

- [x] Update a bucket

- [x] Query data

- [x] Write data 

- [x] Analyze a Flux query

- [x] Generate an Abstract Syntax Tree (AST) from a query

- [x] Retrieve query suggestions

- [x] Retrieve query suggestions for a branching suggestion

**Authentication options**

Use the Token authentication scheme to authenticate to the InfluxDB API.

**Fields**

Host
Port
Protocol
API token

### If the feature is approved, would you be willing to submit a PR?
Yes .

### PR DESCRIPTION
**Pull request by stepinfwd**
resolves #2564

### CODE REVIEW

File: plugins/packages/influxdb/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    try {
+      switch (operation) {
+        case 'list_buckets': {
+          response = await got(`${protocol}://${host}:${port}/api/v2/buckets`, {
+            method: 'get',
+            headers: authHeader(apiKey),
+          });
+
+          result = JSON.parse(response.body);
```
Comment (gsmithun4) [inline plugins/packages/influxdb/lib/index.ts:28]: Use `this.parseJSON`

File: plugins/packages/influxdb/lib/manifest.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  "source": {
+    "name": "InfluxDB",
+    "kind": "influxdb",
+    "exposedVariables": {
+      "isLoading": false,
+      "data": {},
+      "rawData": {},
+      "customTesting": false
+    },
+    "options": {
```
Comment (gsmithun4) [inline plugins/packages/influxdb/lib/manifest.json:15]: `type` is not required in `options`
```
      "api_token": {
        "encrypted": true
      },
```
This is enough

File: plugins/packages/oracledb/package.json [hunk 3]
```diff
@@ -21,8 +21,5 @@
     "knex": "^0.95.14",
     "oracledb": "^5.3.0",
     "react": "^17.0.2"
-  },
-  "devDependencies": {
-    "@types/oracledb": "^5.2.2"
```
Comment (gsmithun4) [inline plugins/packages/oracledb/package.json:26]: Please revert the other plugin changes, if required add it in separate PR

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: plugins/packages/influxdb/lib/manifest.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      "api_token": {
+        "encrypted": true
+      },
+      "port": {
```
Comment (gsmithun4) [inline plugins/packages/influxdb/lib/manifest.json:19]: You can remove `port`, `host` and `protocol`

File: docs/yarn.lock [hunk 5]
```diff
@@ -2,280 +2,178 @@
 # yarn lockfile v1
```
Comment (gsmithun4) [inline docs/yarn.lock:2]: Are these changes expected?

**Review by gsmithun4 — APPROVED** [review]

**Review by Navaneeth-pk — APPROVED** [review]

**Pull request merged**